core/src/main/java/org/hibernate/ogm/transaction/emulated/impl/EmulatedLocalTransactionCoordinatorBuilder.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.transaction.emulated.impl;
+
+import org.hibernate.ConnectionAcquisitionMode;
+import org.hibernate.ConnectionReleaseMode;
+import org.hibernate.ogm.transaction.impl.ForwardingTransactionCoordinatorOwner;
+import org.hibernate.resource.transaction.TransactionCoordinator;
+import org.hibernate.resource.transaction.TransactionCoordinatorBuilder;
+import org.hibernate.resource.transaction.backend.jdbc.spi.JdbcResourceTransaction;
+import org.hibernate.resource.transaction.backend.jdbc.spi.JdbcResourceTransactionAccess;
+import org.hibernate.resource.transaction.spi.TransactionCoordinatorOwner;
+import org.hibernate.resource.transaction.spi.TransactionStatus;
+
+/**
+ * Builds non-JTA-based {@link TransactionCoordinator}s which use a no-op {@link JdbcResourceTransaction} and
+ * connection. That way flushes are triggered as expected - and written to the datastore, but no actual transaction is
+ * involved.
+ *
+ * @author Gunnar Morling
+ */
+public class EmulatedLocalTransactionCoordinatorBuilder implements TransactionCoordinatorBuilder {
```

Comment:
It works as is, AFAICS. It actually even did without the custom `JdbcResourceTransaction`, I added this basically "just to be sure". Do you see anything specific you deem problematic?